### ISSUE TITLE
Install request: NWChem 6.6

### ISSUE THREAD

**Issue by heatherkellyucl** [UCL-RITS/rcps-buildscripts#28]
Legion install requested. IN00874137

**Comment by heatherkellyucl** [comment]
Started 6.6 r27746 building.

**Comment by heatherkellyucl** [comment]
Build process failed on Legion with 
```
/dev/shm/tmp.VB3DpmjULc/nwchem-6.6/src/tools/install/lib/libarmci.a(openib.o): In function `armci_client_con
nect_to_servers':
../../ga-5-4/armci/src/devices/openib/openib.c:(.text+0x2764): undefined reference to `ibv_modify_qp'
../../ga-5-4/armci/src/devices/openib/openib.c:(.text+0x285e): undefined reference to `ibv_modify_qp'
../../ga-5-4/armci/src/devices/openib/openib.c:(.text+0x2938): undefined reference to `ibv_modify_qp'
```
and on Grace with

```
ifort  -c -i8 -align -fpp -qopt-report-file=stderr -fimf-arch-consistency=true -finline-limit=250 -O3  -unroll  -ip -xHost -I. -I../include -I/dev/shm/tmp.vLuV8O36sv/nwchem-6.6/src/include -I/dev/shm/tmp.vLuV8O36sv/nwchem-6.6/src/tools/install/include -DEXT_INT -DLINUX -DLINUX64 -DIFCV8 -DIFCLINUX -DSCALAPACK -DPARALLEL_DIAG -DDEBUG_PRINT  ccsd_t_kernels_omp.F
/tmp/ifortxwkSof.i: catastrophic error: **Internal compiler error: segmentation violation signal raised** Please report this error along with the circumstances in which it occurred in a Software Problem Report.  Note: File and line given may not be explicit cause of this error.
compilation aborted for ccsd_t_kernels_omp.F (code 1)
make[3]: *** [/dev/shm/tmp.vLuV8O36sv/nwchem-6.6/lib/LINUX64/libtce.a(ccsd_t_kernels_omp.o)] Error 1
```

Build script needs modifying for 6.6.

**Comment by heatherkellyucl** [comment]
The Legion attempt failed to link ScaLAPACK correctly, so it looks like they've changed that. There are a bunch of patches for the current NWChem version and it appears that some of them should deal with that.

**Comment by heatherkellyucl** [comment]
Patched install has worked on Legion, have submitted example job. (And building on Grace).

**Comment by heatherkellyucl** [comment]
8 core test example ran successfully on Legion and with similar results to last time.

**Comment by heatherkellyucl** [comment]
The patched NWChem is still failing on Grace.

```
offl_ccsd_t_doubles_l.F(1813,25):remark #34026: call to memcpy implemented as a call to optimized library version
===========================================================================

Begin optimization report for: OFFL0_CCSD_T_DOUBLES_L_12

    Report from: Interprocedural optimizations [ipo]

INLINE REPORT: (OFFL0_CCSD_T_DOUBLES_L_12) [22] offl_ccsd_t_doubles_l.F(1857,19)

===========================================================================
ifort  -c -i8 -align -fpp -qopt-report-file=stderr -fimf-arch-consistency=true -finline-limit=250 -O3  -unroll  -ip -xHost -I. -I../include -I/dev/shm/tmp.Dqf0rU68NX/nwchem-6.6/src/include -I/dev/shm/tmp.Dqf0rU68NX/nwchem-6.6/src/tools/install/include -DEXT_INT -DLINUX -DLINUX64 -DIFCV8 -DIFCLINUX -DSCALAPACK -DPARALLEL_DIAG -DDEBUG_PRINT  ccsd_t_kernels_omp.F
/tmp/ifortrW3XxH.i: catastrophic error: **Internal compiler error: segmentation violation signal raised** Please report this error along with the circumstances in which it occurred in a Software Problem Report.  Note: File and line given may not be explicit cause of this error.
compilation aborted for ccsd_t_kernels_omp.F (code 1)
make[3]: *** [/dev/shm/tmp.Dqf0rU68NX/nwchem-6.6/lib/LINUX64/libtce.a(ccsd_t_kernels_omp.o)] Error 1
```

**Comment by heatherkellyucl** [comment]
From this thread http://www.nwchem-sw.org/index.php/Special:AWCforum/st/id1649/compilation_problem_with_nwchem-....html it looks like ccsd_t_kernels_omp.F requires -openmp in the compile line in some cases (and it did compile that file successfully when I tried). Can be set with `USE_OPENMP=y` at the start. 

I'm not sure what else this affects in the build.

**Comment by heatherkellyucl** [comment]
It built fine on Grace with `USE_OPENMP=y` set.

**Comment by heatherkellyucl** [comment]
Available as `nwchem/6.6-r27746/intel-2015-update2` module.
